src/bitlocker_fmt_plug.c

```diff
@@ -0,0 +1,280 @@
+/* JtR format to crack BitLocker hashes.
+ *
+ * This software is Copyright (c) 2017, Dhiru Kholia <kholia at kth.se> and it
+ * is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ * Big thanks to Joachim Metz and Elena Ago for making this format possible.
+ *
+ * http://jessekornblum.com/publications/di09.pdf (Implementing BitLocker Drive
+ * Encryption for Forensic Analysis) by Jesse D. Kornblum is a useful reference.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_bitlocker;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_bitlocker);
+#else
+
+#include <string.h>
+#include <assert.h>
+#include <errno.h>
+#ifdef _OPENMP
+#include <omp.h>
+#ifndef OMP_SCALE
+#define OMP_SCALE               1
+#endif
+#endif
+
+#include "arch.h"
+#include "misc.h"
+#include "common.h"
+#include "formats.h"
+#include "params.h"
+#include "unicode.h"
+#include "options.h"
+#include "johnswap.h"
+#include "aes.h"
+#include "aes_ccm.h"
+#include "sha.h"
+#include "sha.h"
+#include "jumbo.h"
+#include "memdbg.h"
+#include "bitlocker_common.h"
+
+#define FORMAT_LABEL            "BitLocker"
+#if ARCH_BITS >= 64
+#define ALGORITHM_NAME          "Iterated-SHA512 AES 64/" ARCH_BITS_STR
+#else
+#define ALGORITHM_NAME          "Iterated-SHA512 AES 32/" ARCH_BITS_STR
+#endif
```

Comment:
I thought PBKDF2 was used but apparently it isn't. But it's definitely SHA-256 and not SHA-512.